src/main/java/com/cs203/smucode/handlers/EventHandler.java

```diff
@@ -0,0 +1,72 @@
+package com.cs203.smucode.handlers;
+
+import com.cs203.smucode.constants.NotificationType;
+import com.cs203.smucode.models.Notification;
+import com.cs203.smucode.services.INotificationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class EventHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(EventHandler.class);
+    private final INotificationService notificationService;
+
+    @Autowired
+    public EventHandler(INotificationService notificationService) {
+        this.notificationService = notificationService;
+    }
+
+    /**
+     * Method to handle incoming event
+     *
+     * @param notification incoming notification
+     */
+    public void handleEvent(Notification notification) {
+        logger.info("Received event: {}", notification);
+
+        // Save event in notifications DB
+        notificationService.createNotification(notification);
+
+        // Notify recipients about the tournament start
+        List<String> recipients = notification.getRecipients();
+        for (String recipient : recipients) {
+            logger.info("Sending event: {} to user: {}", notification, recipient);
+            sendNotification(
+                    recipient,
+                    notification.toString()
```

Comment:
i think we need to model another dto, crrect me if im wrong but this will tostring the entire object,
so we are giving all fields and some are not needed (recipients, etc)
for our frontend, is only concerned with type, tournamentId, tournament name, createdAt, isRead

Reply:
yeah ure right, can create a DTO to be send to clients

Reply:
daddys always right.